src/libexec/zmqstat

```diff
@@ -88,7 +102,7 @@ sub getHashPath {
 
 sub processQ {
 	if (! -f $_ ) {return;}
-	if (! m{(^|/)[A-F0-9]{6,}$} ) { return; }
+	if (! m{(?:^|/)(?:[A-F0-9]{6,}|[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-z]{10}z[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-y]+)$} ) { return; }
```

Comment:
Well, it isn't.
`z` acts as separator between the 10-character portion which is derived from seconds and microseconds and the one representing the inode number. And the latter portion cannot contain the character `z`

Reply:
Per [http://www.postfix.org/postconf.5.html#enable_long_queue_ids](http://www.postfix.org/postconf.5.html#enable_long_queue_ids)

> New queue files are created with names such as 3Pt2mN2VXxznjll. These are encoded in a 52-character alphabet that contains digits (0-9), upper-case letters (B-Z) and lower-case letters (b-z). For safety reasons the vowels (AEIOUaeiou) are excluded from the alphabet. The name format is: 6 or more characters for the time in seconds, 4 characters for the time in microseconds, the 'z'; the remainder is the file inode number encoded in the first 51 characters of the 52-character alphabet.

So while they aren't now, it is documented that they can be.  The code should handle that.  Given the fact that the 'z' won't show up in the second part, we could remove the '?', but leaving it ensures we don't get tripped up if it does happen to show up.

Reply:
Ok. I think the format forbids the `z` in the last portion because, given the variable length of the first time related portion, it's the only way to distinguish the part related to time to the one related to the inode number. In fact, if you noticed, the position of the last literal `z` is what's used in the algorithm for the queue directory format to find the microsecond portion of the QID, which I used in the other related commit. And that's lifted straight from the postfix code. So I don't think they are gonna change that in the future. What's you opinion?

Reply:
It sounds like we're perhaps in agreement?  Something like (removing the '?' due to the fact that z should not appear in the "remainder" (inode encoded):
```
if (! m{(?:^|/)(?:[A-F0-9]{6,}|$RECHR{10,}z$RECHR+)$} ) { return; }
```

Reply:
@plobbes Do you reckon this would be more readable, if less compact, to use across all files as definitions at the top?
I kept the exclusion of "z" in the inode number part, since the only purpose of the "z" literal separator is to identify the boundary of the two variable-length portions preceding it and following it (they backtrack 4 characters from its position to get the microseconds portion for instance), therefore I find rather unlikely that they might lift the constraint on its absence in the inode number portion.
```perl
# Postfix QID regexes

# Short Format character: ASCII uppercase A-F range plus ASCII digits
my $SF_QID_CHAR = qr{[A-F0-9]}

# Long Format time portion character:  ASCII digits and ASCII uppercase/lowercase consonants 
my $LF_QID_TIME_CHAR  = qr{[0-9BCDFGHJKLMNPQRSTVWXYZ]}i
# Long Format inode number portion character: ASCII digits and ASCII uppercase/lowercase consonants minus "z"
my $LF_QID_INODE_CHAR = qr{[0-9BCDFGHJKLMNPQRSTVWXYZbcdfghjklmnpqrstvwxy]}

my $REGEX_POSTFIX_QID = qr{(?:${SF_QID_CHAR}{6,}|${LF_QID_TIME_CHAR}{10,}z${LF_QID_INODE_CHAR}+)};
```